### PR TITLE
Dotnet swap out public build for local build in integration tests

### PR DESCRIPTION
**Pull request by eatonphil**

### CODE REVIEW

File: src/clients/dotnet/docs.zig [hunk 1]
```diff
@@ -13,14 +17,80 @@ fn current_commit_pre_install_hook(
...
+
+    const public_reference =
+        \\  <ItemGroup>
+        \\    <PackageReference Include="tigerbeetle" Version="0.13.88" />
+        \\  </ItemGroup>
+    ;
+    const old_csproj_contents = try read_file(arena, csproj_filename);
+    assert(std.mem.containsAtLeast(u8, old_csproj_contents, 1, public_reference));
+
+    const local_reference = try std.fmt.allocPrint(
```
Comment (batiati) [inline src/clients/dotnet/docs.zig:93]: Can't we use the dotnet cli for that?

https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-add-reference

Comment (eatonphil) [reply src/clients/dotnet/docs.zig:93]: Yep, swapped out for that. Thanks!

File: src/clients/shutil.zig [hunk 2]
```diff
@@ -241,3 +243,14 @@ pub fn write_shell_newlines_into_single_line(
     // The above commands all end with ` {cmd_sep} `
     try into.appendSlice("echo ok");
 }
+
+pub fn read_file(arena: *std.heap.ArenaAllocator, file_name: []const u8) ![]const u8 {
```
Comment (matklad) [inline src/clients/shutil.zig]: I think this is already available in std as  `std.fs.cwd().readFileAlloc`

File: src/clients/dotnet/README.md [hunk 3]
```diff
@@ -446,3 +446,4 @@ $ dotnet clean --verbosity minimal
 $ dotnet build
 $ if ($env:TEST -eq 'true') { dotnet test } else { echo "Skipping client unit test" }
 ```
```
Comment (matklad) [inline src/clients/dotnet/README.md]: might want to rebase this on main, as we normalize whitespace there now.

File: src/clients/shutil.zig (hunk 2 above)
Comment (eatonphil) [reply src/clients/shutil.zig]: Sweet, thank you.

File: src/clients/dotnet/README.md (hunk 3 above)
Comment (eatonphil) [reply src/clients/dotnet/README.md]: Thanks!

File: src/clients/dotnet/TigerBeetle/Client.cs [hunk 4]
```diff
@@ -132,4 +131,4 @@ private void Dispose(bool disposing)
             nativeClient.Dispose();
         }
     }
-}
+}
```
Comment (matklad) [inline src/clients/dotnet/TigerBeetle/Client.cs]: needs trailing newline

Comment (eatonphil) [reply src/clients/dotnet/TigerBeetle/Client.cs]: It was a change to trigger failures, not intended to be committed.

**Review by batiati — DISMISSED** [review]

File: src/clients/dotnet/docs.zig [hunk 5]
```diff
@@ -1,24 +1,137 @@
...
+        return error.CSProjFileNotFound;
     };
+    assert(file_or_directory_exists(csproj_filename));
+
+    try run_shell(arena, "dotnet remove package tigerbeetle");
+
+    const public_reference =
+        \\</Project>
+    ;
+    const old_csproj_contents = try std.fs.cwd().readFileAlloc(arena.allocator(), csproj_filename, 1024 * 10);
```
Comment (batiati) [inline src/clients/dotnet/docs.zig]:
```suggestion
    const old_csproj_contents = try std.fs.cwd().readFileAlloc(arena.allocator(), csproj_filename, std.math.maxInt(usize));
```

File: src/clients/dotnet/docs.zig [hunk 6]
```diff
@@ -1,24 +1,137 @@
...
+        ),
+    );
+
+    try std.os.chdir(sample_directory);
+    // Find the .csproj file so we can swap out the public package
+    // with our local build, if the .csproj file exists.
+    var dir = try std.fs.cwd().openIterableDir(".", .{});
+    defer dir.close();
+
+    var walker = try dir.walk(arena.allocator());
```
Comment (batiati) [inline src/clients/dotnet/docs.zig]: Can we move the `walker` inside the block where it is used?

Comment (eatonphil) [reply src/clients/dotnet/docs.zig]: Yep thank you, got pushed far away accidentally.

**Review by batiati — APPROVED** [review]

**Pull request merged**